keps/sig-node/2040-kubelet-cri/README.md

```diff
@@ -0,0 +1,270 @@
+- [Implementation History](#implementation-history)
+<!-- /toc -->
+
+## Release Signoff Checklist
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [x] (R) KEP approvers have approved the KEP status as `implementable`
+- [x] (R) Design details are appropriately documented
+- [x] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input
+- [x] (R) Graduation criteria is in place
+- [x] (R) Production readiness review completed
+- [ ] Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+Identify remaining gaps to promote CRI to Beta and GA to reflect its practical use in production for many years.
+
+## Motivation
+CRI based runtimes such as CRI-O and containerd have been in use in production for over a year now with the current CRI API.
+We want to signal to the users that the CRI API is production ready and they should feel comfortable moving away dockershim
+as it is slated to be deprecated.
+
+### Goals
+- Graduate the CRI API to stable.
+- Identify any fields that need to made more type safe such as Seccomp.
+- Address and cleanup the notes/todos in the CRI.
+
+### Non-Goals
+- Block on any big new features.
+
+## Proposal
+Evolve the CRI API version as we address feedback in each milestone towards stable.
+- v1alpha2 (alpha, current state)
+- v1beta (beta, proposed 1.20)
+- v1 (stable, TBD)
+
+### Risks and Mitigations
+| Risk  | Detail  | Mitigation  |
+|---|---|---|
+| CRI stats performance  | CRI stats performance may be worse compared to cadvisor  |  Measure performance and share report with community |
+
+## Design Details
+
+### Identified Work Items
+- No longer map the `container-runtime-endpoint` flag as experimental.
+- Keep the `image-service-endpoint` flag as experimental and evaluate if it makes sense to keep
+  as a configurable or remove it.
+
+#### Changes from v1alpha2 to v1beta1
+
+- kubenet: There exists an open TODO in the specification to remove support for setting PodCidr for kubenet networking. However for CRI
+implementations CNI is the existing standard and is primarily the only solution being tested with the CRI container runtime integrations.
+Need Sig-Node and Sig-Networking to help validate if / when kubenet is being deprecated and if we should deprecate this before beta. If not when. 
```

Comment:
I believe it is not deprecated by SIG network yet. cc/ @freehan from SIG network.

Reply:
Yes. Kubenet is not deprecated yet. @kubernetes/sig-network-misc